nur/index.py

```diff
@@ -7,7 +7,56 @@
 from typing import Any, Dict
 
 
-def index_repo(directory: Path, repo: str, expression_file: str) -> Dict[str, Any]:
+def resolve_source(pkg: Dict, repo: str, url: str) -> str:
+    # TODO commit hash
+    prefix = f"https://github.com/nix-community/nur-combined/tree/master/repos/{repo}"
+    position = pkg["meta"].get("position", None)
+    if position is not None and position.startswith("/nix/store"):
+        path_str, line = position.rsplit(":", 1)
+        path = Path(path_str)
+        # I've decided to just take these 2 repositories,
+        # update this whenever someone decided to use a recipe source other than
+        # NUR or nixpkgs to override packages on. right now this is about as accurate as
+        # `nix edit` is
+        # TODO find commit hash
+        prefixes = {
+            "nixpkgs": "https://github.com/nixos/nixpkgs/tree/master/",
+            "nur": "https://github.com/nix-community/nur-combined/tree/master/",
+        }
+        stripped = path.parts[4:]
+        if path.parts[3].endswith("source"):
+
+            canonical_url = url
+            # if we want to add the option of specifying branches, we have to update this
+            if "github" in url:
+                canonical_url += "/blob/master/"
+            elif "gitlab" in url:
+                canonical_url += "/-/blob/master/"
```

Comment:
Also works on gitlab: https://gitlab.redox-os.org/redox-os/relibc/-/blob/HEAD/README.md

```suggestion
                canonical_url += "/-/blob/HEAD/"
```